### DISCUSSION TITLE
[Question] rendered html has `nohighlight` class. Is this on purpose?

### DISCUSSION THREAD

**Discussion by i10416** [planet42/Laika#239] [Q&A]
Is this intentional?

```scala
if (hasHighlighting) Style.noHighlight else Styles(language)
```

https://github.com/planet42/Laika/blob/master/core/shared/src/main/scala/laika/render/HTMLRenderer.scala#L117

I think it makes sense if code block with syntax highlighting is rendered with language class(e.g. `class="scala"`) and code block without syntax highlighting is rendered with `nohighlight` class.

![image](https://user-images.githubusercontent.com/39330037/138847415-600ce1b1-afb2-4416-9857-832cf514bdf4.png)

![image](https://user-images.githubusercontent.com/39330037/138847982-70f95da6-0397-4c98-8ee8-089231f4069f.png)

**Comment by i10416** [comment]
Ah, I understood. "nohighlight" is for `highlight.js`, isn't it?

https://planet42.github.io/Laika/0.18/03-preparing-content/05-syntax-highlighting.html#integrating-external-tools

**Comment by jenshalm** [comment]
Yes, this is intentional, to allow for a combination of internal and external highlighters on the same page. Is there a scenario where this is problematic?

**Comment by i10416** [comment]
@jenshalm No, there's no problem for now : )  Thanks!

Btw, I found demo site does not look nice on mobile or smaller screen, so I created similar demo frontend without js!
(using Scala.js, [Laminar](https://laminar.dev/) and [scalacss](https://japgolly.github.io/scalacss/book/index.html))

Live demo: https://i10416.github.io/demo/ 
Note that current site is not optimized yet and thus, first load is slow 😓

I think Laminar is type-safe and easier to use (for Scala developers) frontend framework native to Scala.
What do you think  if I suggest  replacing the demo site with Laminar's one?

Source code of demo site above is at https://github.com/i10416/laika-laminar.

**Comment by jenshalm** [comment]
@i10416: I just opened the discussion section for this repository and moved all issues with label 'question' from the last 2 years here. 

Regarding changing the demo app: I think improving the UI for mobile and moving to Laminar are two concerns I'd prefer to keep separate. I can look into mobile improvements during the 0.18 timeline, and then later into moving to Laminar. It's certainly an attractive option as it would simplify the build. But I need to learn a thing or two about Laminar before making such a big change, so that I'm able to maintain it myself when needed. I think I'd also like to maintain some of the current look and feel.

Another thing I'm thinking about is adding a showcase section to the manual. I could link to your alternative demo app there until we change it in this repo.

**Reply by i10416** [reply]
> I think improving the UI for mobile and moving to Laminar are two concerns I'd prefer to keep separate. 

Yes, I agree. There would be a lot of changes.

Showcase section sounds nice!  Realworld examples like https://http4s.org will encourage developers to adopt Laika!
